src/modules/blocks/nested-blocks.js

```diff
@@ -0,0 +1,37 @@
+import { blockToTable, replaceElement } from '../utils/dom-utils.js';
+
+/**
+ * retroactively converts any nested blocks into tables.
+ * hlx converter does not handle nested blocks, for good reason. but we do have use-cases where block are nested
+ * For that, this conversion turns only nested blocks into tables, and those tables will be handled at the frontend.
+ * @param {Document} document
+ */
+export default function handleNestedBlocks(document) {
+  // Add container block classes this array
+  const blockClasses = [
+    'shade-box',
+    'table',
+    'tabs',
+    'code',
+    'note',
+    'related-articles',
+    'Badge',
+    'embed',
+  ];
+
+  // top level blocks (container blocks) blocks have the strict structure body > main > div > div
+  const blocks = Array.from(
+    document.querySelectorAll('body > main > div > div'),
+  );
+
+  // find nested blocks (converted in previous steps), and convert it to a table
+  blocks.forEach((block) => {
```

Comment:
there seems to be an assumption (I think a good one) that blocks can only be nested one deep. However, I'd be a bit concerned with some of these blocks that that will not hold. especially with things like tabs/accordions/etc. I suspect you'll fin cases where that content needs to contain other blocks. 

Hopefully I'm wrong about that, but I think allowing deeper nesting would be problematic.


All of this is to say: I think for blocks where 1 deep nesting won't work, we'll need to reconsider how those work. I can show you some examples of how we do tabs/accordions, etc. in other docx authored Franklin sites (hint: sections), and for those we may want to follow that approach.


May be we can ignore this for now, but just keep that in mind for the future if/when you find you need to go more than 2 deep.

Reply:
I made an assumption for one level of nesting and figured if we need to handle multiple levels, we can handle in subsequent PR. I haven't seen deeper nesting yet, but I'd guess there might content that does have deeper nesting.

I'd think that we should handle deep nesting, if present in content, regardless.

I would agree that deeper levels of nesting are problematic. However, we can solve it in two ways, either:

1. make the decision to update the content source (markdown docs)
2. or we handle any level of nesting as we might not know which pages have nesting and how deep.